include/bm/bm_sim/deparser.h

```diff
@@ -23,18 +23,42 @@
 #ifndef BM_BM_SIM_DEPARSER_H_
 #define BM_BM_SIM_DEPARSER_H_
 
+#include <bm/bm_sim/debugger.h>
+#include <bm/bm_sim/logger.h>
+#include <bm/bm_sim/actions.h>
+#include <bm/bm_sim/packet.h>
+#include <bm/bm_sim/phv.h>
+
 #include <vector>
 #include <string>
+#include <mutex>
 
 #include "named_p4object.h"
 #include "phv_forward.h"
+#include "stateful.h"
 
 namespace bm {
 
 class Packet;
 
 class Checksum;
 
+struct DeparserOp {
```

Comment:
DeparserOp cant be moved to deparser.cpp because it is type argument to unique_ptr and the type argument to unique_ptr must be complete type (forward declared struct is incomplete). With DeparserOpMethodCall there is no problem so I moved it.

Reply:
This is not true. The destructor definition needs the complete type declaration, so the destructor needs to be defined in the cpp file, but it is perfectly fine to forward-declare the type argument to `unique_ptr`. This is actually used very often in modern C++ when defining PIMPL, with a smart pointer for the implementation member. You can Google "PIMPL with unique_ptr". See for example: https://stackoverflow.com/questions/9020372/how-do-i-use-unique-ptr-for-pimpl